opentelemetry-instrumentation/src/opentelemetry/instrumentation/_semconv.py

```diff
@@ -0,0 +1,217 @@
+_SPAN_ATTRIBUTES_NETWORK_PEER_ADDRESS = "network.peer.address"
+_SPAN_ATTRIBUTES_NETWORK_PEER_PORT = "network.peer.port"
+_METRIC_ATTRIBUTES_CLIENT_DURATION_NAME = "http.client.request.duration"
+
+_client_duration_attrs_old = [
+    SpanAttributes.HTTP_STATUS_CODE,
+    SpanAttributes.HTTP_HOST,
+    SpanAttributes.NET_PEER_PORT,
+    SpanAttributes.NET_PEER_NAME,
+    SpanAttributes.HTTP_METHOD,
+    SpanAttributes.HTTP_FLAVOR,
+    SpanAttributes.HTTP_SCHEME,
+]
+
+_client_duration_attrs_new = [
+    _SPAN_ATTRIBUTES_ERROR_TYPE,
+    SpanAttributes.HTTP_REQUEST_METHOD,
+    SpanAttributes.HTTP_RESPONSE_STATUS_CODE,
+    SpanAttributes.NETWORK_PROTOCOL_VERSION,
+    SpanAttributes.SERVER_ADDRESS,
+    SpanAttributes.SERVER_PORT,
+    # TODO: Support opt-in for scheme in new semconv
+    # SpanAttributes.URL_SCHEME,
+]
+
+
+def _filter_duration_attrs(attrs, sem_conv_opt_in_mode):
+    filtered_attrs = {}
+    allowed_attributes = (
+        _client_duration_attrs_new
+        if sem_conv_opt_in_mode == _OpenTelemetryStabilityMode.HTTP
+        else _client_duration_attrs_old
+    )
+    for key, val in attrs.items():
+        if key in allowed_attributes:
+            filtered_attrs[key] = val
+    return filtered_attrs
+
+
+def set_string_attribute(result, key, value):
+    if value:
+        result[key] = value
+
+
+def set_int_attribute(result, key, value):
+    if value:
+        try:
+            result[key] = int(value)
+        except ValueError:
+            return
+
+
+def _set_http_method(result, original, normalized, sem_conv_opt_in_mode):
+    original = original.strip()
+    normalized = normalized.strip()
+    # See https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#common-attributes
+    # Method is case sensitive. "http.request.method_original" should not be sanitized or automatically capitalized.
+    if original != normalized and _report_new(sem_conv_opt_in_mode):
+        set_string_attribute(
+            result, SpanAttributes.HTTP_REQUEST_METHOD_ORIGINAL, original
+        )
+
+    if _report_old(sem_conv_opt_in_mode):
+        set_string_attribute(result, SpanAttributes.HTTP_METHOD, normalized)
+    if _report_new(sem_conv_opt_in_mode):
+        set_string_attribute(
+            result, SpanAttributes.HTTP_REQUEST_METHOD, normalized
+        )
+
+
+def _set_http_url(result, url, sem_conv_opt_in_mode):
+    if _report_old(sem_conv_opt_in_mode):
+        set_string_attribute(result, SpanAttributes.HTTP_URL, url)
+    if _report_new(sem_conv_opt_in_mode):
+        set_string_attribute(result, SpanAttributes.URL_FULL, url)
+
+
+def _set_http_scheme(result, scheme, sem_conv_opt_in_mode):
+    if _report_old(sem_conv_opt_in_mode):
+        set_string_attribute(result, SpanAttributes.HTTP_SCHEME, scheme)
+    # TODO: Support opt-in for scheme in new semconv
+    # if _report_new(sem_conv_opt_in_mode):
+    #     set_string_attribute(result, SpanAttributes.URL_SCHEME, scheme)
```

Comment:
Similar to above comment. Shouldn't we set this even if that constant doesn't exist so as to follow the new semconv?